.github/scripts/validate-pkgbuilds.js

```diff
@@ -2,14 +2,55 @@ import { readdirSync } from "node:fs";
 import { join } from "node:path";
 import { $ } from "bun";
 
+const mainBranch = "master";
 const pkgbuildsDir = join(import.meta.dir, "..", "..");
 
-const pkgbuildFiles = readdirSync(pkgbuildsDir, {
-  recursive: true,
-  withFileTypes: true,
-})
-  .filter((x) => x.isFile() && x.name === "PKGBUILD")
-  .map((x) => x.parentPath);
+async function getChangedPkgbuilds() {
+  const { stdout } =
+    await $`git diff --name-only ${process.env.BASE_REF} ${process.env.HEAD_REF}`
+      .nothrow()
+      .quiet();
+
+  return stdout
+    .toString("utf8")
+    .trim()
+    .split("\n")
+    .filter(Boolean)
+    .filter((file) => file.endsWith("/PKGBUILD"))
```

Comment:
Just FYI, it can be done a little easier by simply adding a pattern at the end of git command:
https://github.com/ventureoo/PKGBUILDs/blob/main/.github/workflows/repo.yml#L67